client/my-sites/plans-features-main/index.tsx

```diff
@@ -261,6 +262,17 @@ const PlansFeaturesMain = ( {
 		useGetFreeSubdomainSuggestion(
 			paidDomainName || siteTitle || signupFlowUserName || currentUserName
 		);
+
+	// the A/A tests for identifying SRM issue. See peP6yB-11Y-p2
+	// We can use `useExperiment()` and its `isEligible` check to avoid the condition, but it's intentional to use loadExperimentAssignment() here
+	// to be consistent with the other A/A tests.
+	useEffect( () => {
+		if ( flowName === 'onboarding' ) {
+			loadExperimentAssignment( 'calypso_srm_test_plans_page_view_free_plan_modal_view' );
+			loadExperimentAssignment( 'calypso_srm_test_plans_page_view_free_plan_button_click' );
+		}
+	}, [] );
```

Comment:
Good point :) Normally that should be the case given that `flowName` was used inside. However, I intentionally left the dependency list empty here since I wanted to make sure they were loaded only once per entering the plans step no matter the subsequent changes of props. I might be overlooking something though, do you see any potential issue of it?

Reply:
Generally the dependency list is not meant to be a logical executor of effects. Checkout the section ```Don’t Lie to React About Dependencies``` below :)

https://overreacted.io/a-complete-guide-to-useeffect/#dont-lie-to-react-about-dependencies.

Also we (Local-storage) cache our experiment fetch, so there is actually no penalty to calling this function repeatedly with the same value :)